aiida/engine/processes/process.py

```diff
@@ -684,27 +685,22 @@ def _setup_db_record(self) -> None:
             elif isinstance(self.node, orm.WorkflowNode):
                 self.node.base.links.add_incoming(parent_calc, LinkType.CALL_WORK, self.metadata.call_link_label)
 
-        self._setup_metadata()
+        self._setup_metadata(copy.copy(dict(self.inputs.metadata)))
```

Comment:
The `CalcJob`, for example, defines the `computer` and `options` keys. It needs to be able to pop those, otherwise this line https://github.com/aiidateam/aiida-core/blob/9d903ca2c73ef0e7e967c17ff3d7eac3410f6b58/aiida/engine/processes/process.py#L709
will raise in the `Process._setup_metadata` step.